### ISSUE TITLE
Internet Access Behind a proxy fails with "The token supplied to the function is invalid"

### ISSUE THREAD

**Issue by wordeater** [PowerShell/PowerShell#16908]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

The Invoke-Webrequest and Invoke-RestMethod cmdlets are generating an error when I try to use them through our web proxy.
If I bypass the proxy completely, there is no error message.
The error happens in both Windows 10 20H2 and Windows Server 2019 1809, which are the two environments I have available.
I also have PowerShell Core 7.2.1 on a Linux system, but it does not go through the proxy and does not get the error.
PowerShell 5.1 does not get the error. In fact, I am running scripts with PowerShell 5.1 instead of PowerShell Core 7.2.1 because it does not get the error.
The steps to reproduce are simple in my environment.
Run PowerShell Core.
PS D:\Tools\Code\dev> Invoke-WebRequest https://www.google.com
Invoke-WebRequest: The token supplied to the function is invalid

I've rebooted both systems on which I test and that did not resolve the issue.
I am able to get to the Internet using Chrome, Edge, and Firefox on these systems.
I am able to get to the Internet using PowerShell 5.1 on these systems.
I use this to pull my proxy credentials into PowerShell 5.1 and PowerShell Core 7.2.1.
(New-Object System.Net.WebClient).Proxy.Credentials = [System.Net.CredentialCache]::DefaultNetworkCredentials

A packet capture and the proxy logs show a successful NTLM authentication from PowerShell and then nothing else after that.
The proxy is a McAfee Web Gateway.

### Expected behavior

```console
PS D:\tools\code\dev> Invoke-WebRequest https://www.google.com


StatusCode        : 200
StatusDescription : OK
Content           : <!doctype html><html itemscope="" itemtype="http://schema.org/WebPage" lang="en"><head><meta content="Search the world's information, including webpages, images, videos and more. Google has many speci...
RawContent        : HTTP/1.1 200 OK
                    Vary: Accept-Encoding
                    Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000...
Forms             : {f}
Headers           : {[Vary, Accept-Encoding], [Alt-Svc, h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"],
                    [Connection, Keep-Alive], [Date, Wed, 23 Feb 2022 15:30:20 GMT]...}
Images            : {@{innerHTML=; innerText=; outerHTML=<IMG id=hplogo style="PADDING-BOTTOM: 14px; PADDING-TOP: 28px; PADDING-LEFT: 0px; PADDING-RIGHT: 0px" alt=Google
                    src="/images/branding/googlelogo/1x/googlelogo_white_background_color_272x92dp.png" width=272 height=92>; outerText=; tagName=IMG; id=hplogo; style=PADDING-BOTTOM: 14px; PADDING-TOP: 28px; PADDING-LEFT: 0px;
                    PADDING-RIGHT: 0px; alt=Google; src=/images/branding/googlelogo/1x/googlelogo_white_background_color_272x92dp.png; width=272; height=92}}
InputFields       : {@{innerHTML=; innerText=; outerHTML=<INPUT type=hidden value=en name=hl>; outerText=; tagName=INPUT; type=hidden; value=en; name=hl}, @{innerHTML=; innerText=; outerHTML=<INPUT type=hidden value=hp
                    name=source>; outerText=; tagName=INPUT; type=hidden; value=hp; name=source}, @{innerHTML=; innerText=; outerHTML=<INPUT type=hidden name=biw>; outerText=; tagName=INPUT; type=hidden; name=biw}, @{innerHTML=;
                    innerText=; outerHTML=<INPUT type=hidden name=bih>; outerText=; tagName=INPUT; type=hidden; name=bih}...}
Links             : {@{innerHTML=<SPAN class=gbtb2></SPAN><SPAN class=gbts>Search</SPAN>; innerText=Search; outerHTML=<A id=gb_1 class="gbzt gbz0l gbp1" href="https://www.google.com/webhp?tab=ww"><SPAN class=gbtb2></SPAN><SPAN
                    class=gbts>Search</SPAN></A>; outerText=Search; tagName=A; id=gb_1; class=gbzt gbz0l gbp1; href=https://www.google.com/webhp?tab=ww}, @{innerHTML=<SPAN class=gbtb2></SPAN><SPAN class=gbts>Images</SPAN>;
                    innerText=Images; outerHTML=<A id=gb_2 class=gbzt href="https://www.google.com/imghp?hl=en&amp;tab=wi"><SPAN class=gbtb2></SPAN><SPAN class=gbts>Images</SPAN></A>; outerText=Images; tagName=A; id=gb_2;
                    class=gbzt; href=https://www.google.com/imghp?hl=en&amp;tab=wi}, @{innerHTML=<SPAN class=gbtb2></SPAN><SPAN class=gbts>Maps</SPAN>; innerText=Maps; outerHTML=<A id=gb_8 class=gbzt
                    href="https://maps.google.com/maps?hl=en&amp;tab=wl"><SPAN class=gbtb2></SPAN><SPAN class=gbts>Maps</SPAN></A>; outerText=Maps; tagName=A; id=gb_8; class=gbzt;
                    href=https://maps.google.com/maps?hl=en&amp;tab=wl}, @{innerHTML=<SPAN class=gbtb2></SPAN><SPAN class=gbts>Play</SPAN>; innerText=Play; outerHTML=<A id=gb_78 class=gbzt
                    href="https://play.google.com/?hl=en&amp;tab=w8"><SPAN class=gbtb2></SPAN><SPAN class=gbts>Play</SPAN></A>; outerText=Play; tagName=A; id=gb_78; class=gbzt; href=https://play.google.com/?hl=en&amp;tab=w8}...}
ParsedHtml        : System.__ComObject
RawContentLength  : 53342
```

### Actual behavior

```console
PS D:\Tools\Code\dev> Invoke-WebRequest https://www.google.com
Invoke-WebRequest: The token supplied to the function is invalid
```

### Error details

```console
Exception             :
    Type            : System.ComponentModel.Win32Exception
    NativeErrorCode : -2146893048
    ErrorCode       : -2147467259
    TargetSite      :
        Name          : GetOutgoingBlob
        DeclaringType : System.Net.NTAuthentication, System.Net.Http, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
        MemberType    : Method
        Module        : System.Net.Http.dll
    Message         : The token supplied to the function is invalid
    Source          : System.Net.Http
    HResult         : -2147467259
    StackTrace      :
   at System.Net.NTAuthentication.GetOutgoingBlob(Byte[] incomingBlob, Boolean throwOnError, SecurityStatusPal& statusCode)
   at System.Net.NTAuthentication.GetOutgoingBlob(String incomingBlob)
   at System.Net.Http.AuthenticationHelper.SendWithNtAuthAsync(HttpRequestMessage request, Uri authUri, Boolean async, ICredentials credentials, Boolean isProxyAuth, HttpConnection connection, HttpConnectionPool
connectionPool, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.AuthenticationHelper.SendWithAuthAsync(HttpRequestMessage request, Uri authUri, Boolean async, ICredentials credentials, Boolean preAuthenticate, Boolean isProxyAuth, Boolean doRequestAuth,
HttpConnectionPool pool, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.EstablishProxyTunnelAsync(Boolean async, HttpRequestHeaders headers, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(HttpRequestMessage request)
   at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPoolManager.SendAsyncMultiProxy(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, MultiProxy multiProxy, Uri firstProxy, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource
pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.PowerShell.Commands.WebRequestPSCmdlet.GetResponse(HttpClient client, HttpRequestMessage request, Boolean keepAuthorization)
   at Microsoft.PowerShell.Commands.WebRequestPSCmdlet.ProcessRecord()
   at System.Management.Automation.CommandProcessor.ProcessRecord()
CategoryInfo          : NotSpecified: (:) [Invoke-WebRequest], Win32Exception
FullyQualifiedErrorId : System.ComponentModel.Win32Exception,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
InvocationInfo        :
    MyCommand        : Invoke-WebRequest
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 12
    Line             : Invoke-WebRequest https://www.google.com
    PositionMessage  : At line:1 char:1
                       + Invoke-WebRequest https://www.google.com
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : Invoke-WebRequest
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.1
PSEdition                      Core
GitCommitId                    7.2.1
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iSazonov** [comment]
> Invoke-WebRequest: The token supplied to the function is invalid

It is not PowerShell error. It comes from underlying OS and says the OS has a cryptographic algorithm disabled.
See https://stackoverflow.com/questions/40939031/tls-1-2-the-token-supplied-to-the-function-is-invalid

**Comment by wordeater** [comment]
I have looked at that article previously, as part of my earlier troubelshooting.
If that is the case, why does it work on the same system with PowerShell 5.1 but fail with PowerShell 7.2.1?
I also created a test script that iterated through setting
`$ProtocolList = @( 'SystemDefault'; 'Ssl3'; 'Tls'; 'Tls11'; 'Tls12'; 'Tls13' ); $ProtocolList | Foreach-Object { $protocol = $_; [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::($protocol); Invoke-WebRequest -Uri "https://www.google.com" }`
And it failed for every protocol with the same error. It got a different error with SystemDefault because it didn't understand that protocol, but the rest all say "The token supplied to the function in invalid"
The cryptographic protocols enabled in the registry are
`RSA/SHA256
RSA/SHA384
RSA/SHA1
ECDSA/SHA256
ECDSA/SHA384
ECDSA/SHA1
DSA/SHA1
RSA/SHA512
ECDSA/SHA512`
What is the best way to get a better error?
Would that be a packet capture and study the SSL logs or can PowerShell provide any detailed logging on that aspect of the problem?
